### PR TITLE
Added pipelining example

### PR DESCRIPTION
**Pull request by pveentjer**

### CODE REVIEW

File: pipeline/src/main/java/PipeliningDemo.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        for (int i = 0; i < iterations; i++) {
+            Pipeline pipeline = new Pipeline(depth);
+            for (long k = 0; k < getsPerIteration; k++) {
+                int key = random.nextInt(keyDomain);
+                pipeline.add(map.getAsync(key));
+            }
+            pipeline.results();
+        }
+        long duration = System.currentTimeMillis();
+        System.out.println("Pipedlined with depth:" + depth + ", duration:" + (duration - startMs) + " ms");
```
Comment (vbekiaris) [inline pipeline/src/main/java/PipeliningDemo.java]: typo: "Pipedlined"

File: pipeline/src/main/java/PipeliningDemo.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+import com.hazelcast.client.HazelcastClient;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.IMap;
+import com.hazelcast.core.Pipeline;
+
+import java.util.Random;
+
+public class PipeliningDemo {
```
Comment (vbekiaris) [inline pipeline/src/main/java/PipeliningDemo.java]: A short javadoc explaining what the user is expecting to see in this sample would be good

File: pipeline/src/main/java/PipeliningDemo.java (hunk 1 above)
Comment (pveentjer) [reply pipeline/src/main/java/PipeliningDemo.java]: Done

File: pipeline/src/main/java/PipeliningDemo.java (hunk 2 above)
Comment (pveentjer) [reply pipeline/src/main/java/PipeliningDemo.java]: Done

**Pull request merged**